zerver/lib/actions.py

```diff
@@ -5169,6 +5173,9 @@ def do_create_realm(
         # If the signups stream hasn't been created in the admin
         # realm, don't auto-create it to send to it; just do nothing.
         pass
+
+    create_system_user_groups_for_realm(realm)
```

Comment:
I think this can and should be moved into the `transaction.atomic` block earlier in this function.